File: framework/languages/java/e2e/RegistrationCodec/Client/src/main/java/systems/zlink/e2e/registrationcodec/client/Scenarios/RcB6JsonGoldenScenario.java
package systems.zlink.e2e.registrationcodec.client.Scenarios;

import java.util.Arrays;
import systems.zlink.e2e.registrationcodec.client.Support.ScenarioAssert;
import systems.zlink.e2e.registrationcodec.client.Support.ScenarioContext;
import systems.zlink.e2e.registrationcodec.shared.Contracts;

public final class RcB6JsonGoldenScenario {
    private RcB6JsonGoldenScenario() {
    }

    public static void run(ScenarioContext context) {
        Contracts.JsonGoldenRes result = context.server()
            .post("/codec/json-golden")
            .submit(Contracts.JsonGoldenRes.class)
            .toCompletableFuture()
            .join()
            .body();
        ScenarioAssert.ensure("Ada Lovelace".equals(result.displayName()),
            "RC-B6 display name changed");
        ScenarioAssert.ensure("ready".equals(result.status()), "RC-B6 status changed");
        ScenarioAssert.ensure(result.balance() == -9_223_372_036_854_775_000L,
            "RC-B6 int64 value changed");
        ScenarioAssert.ensure(Arrays.equals(result.payload(),
                new byte[] {0x00, 0x7f, (byte) 0x80, (byte) 0xff}),
            "RC-B6 bytes value changed");
        ScenarioAssert.ensure(result.score() == 2_147_000_001, "RC-B6 int32 value changed");
        ScenarioAssert.ensure(Double.compare(result.ratio(), 0.125) == 0,
            "RC-B6 floating-point value changed");
        ScenarioAssert.ensure(result.optionalNote() == null, "RC-B6 nullable value changed");
        ScenarioAssert.ensure("application/json".equals(result.contentType()),
            "RC-B6 did not use default JSON: " + result.contentType());
        ScenarioAssert.waitForEvidence(context.evidence(), "ContentType", "JsonGoldenReq", "application/json");
        System.out.println("scenario RC-B6 passed");
    }
}

File: framework/languages/java/e2e-kotlin/RegistrationCodec/Client/src/main/kotlin/systems/zlink/e2e/kotlin/registrationcodec/client/Scenarios/RcB6JsonGoldenScenario.kt
package systems.zlink.e2e.kotlin.registrationcodec.client.scenarios

import systems.zlink.e2e.kotlin.registrationcodec.JsonGoldenRes
import systems.zlink.e2e.kotlin.registrationcodec.client.support.ScenarioAssert
import systems.zlink.e2e.kotlin.registrationcodec.client.support.ScenarioHttpClient

class RcB6JsonGoldenScenario(
    private val server: ScenarioHttpClient,
    private val assert: ScenarioAssert,
) {
    fun run() {
        val result = server.post<JsonGoldenRes>("/codec/json-golden")
        assert.that(result.displayName == "Ada Lovelace", "RC-B6 display name changed")
        assert.that(result.status == "ready", "RC-B6 status changed")
        assert.that(result.balance == -9_223_372_036_854_775_000L, "RC-B6 int64 value changed")
        assert.that(
            result.payload.contentEquals(byteArrayOf(0x00, 0x7f, 0x80.toByte(), 0xff.toByte())),
            "RC-B6 bytes value changed",
        )
        assert.that(result.score == 2_147_000_001, "RC-B6 int32 value changed")
        assert.that(result.ratio == 0.125, "RC-B6 floating value changed")
        assert.that(result.optionalNote == null, "RC-B6 nullable value changed")
        assert.that(result.contentType == "application/json", "RC-B6 did not use default JSON")
        assert.that(
            server.waitForEvidence(
                "Request",
                "JsonGoldenReq",
                "Ada Lovelace|ready|-9223372036854775000|2147000001|4",
            ).any { it.marker == "Request" && it.packetName == "JsonGoldenReq" },
            "RC-B6 server evidence missing",
        )
        println("scenario RC-B6 passed")
    }
}
